temporal-sdk/src/main/java/io/temporal/nexus/WorkflowRunOperation.java

```diff
@@ -49,7 +56,32 @@ public OperationStartResult<R> start(
             operationStartDetails.getCallbackUrl(),
             operationStartDetails.getCallbackHeaders(),
             nexusCtx.getTaskQueue());
-    return OperationStartResult.async(handle.getInvoker().invoke(nexusRequest).getWorkflowId());
+
+    WorkflowExecution workflowExec = handle.getInvoker().invoke(nexusRequest);
+
+    // Create the link information about the new workflow and return to the caller.
+    Link.WorkflowEvent workflowEventLink =
+        Link.WorkflowEvent.newBuilder()
+            .setNamespace(nexusCtx.getNamespace())
+            .setWorkflowId(workflowExec.getWorkflowId())
+            .setRunId(workflowExec.getRunId())
+            .setEventRef(
+                Link.WorkflowEvent.EventReference.newBuilder()
+                    .setEventType(EventType.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED))
+            .build();
+    io.temporal.api.nexus.v1.Link nexusLink = workflowEventToNexusLink(workflowEventLink);
+    try {
+      OperationStartResult.Builder<R> result =
+          OperationStartResult.<R>newBuilder().setAsyncOperationId(workflowExec.getWorkflowId());
+      if (nexusLink != null) {
+        result.addLink(nexusProtoLinkToLink(nexusLink));
+      }
+      return result.build();
+    } catch (URISyntaxException e) {
+      // Not expected as the link is constructed by the SDK.
+      throw new OperationHandlerException(
+          OperationHandlerException.ErrorType.BAD_REQUEST, "failed to construct result URL", e);
```

Comment:
Wouldn't this be an internal error since it originated in the handler code?

Reply:
Yeah I think you are right.